src/layer/ProcessingDataFusionLayer.ts

```diff
@@ -95,13 +95,23 @@ export class ProcessingDataFusionLayer extends AbstractSentinelHubV3Layer {
           datasource.dataFilter.mosaickingOrder = layerInfo.layer.mosaickingOrder;
         }
 
+        // note that we should be using updateProcessingGetMapPayload or sth. similar here, this is just a
```

Comment:
For some reason I can't add comments to the unchanged parts of the code.

However, would it work to just run `updateProcessingGetMapPayload` of the `bogusFirstLayer` on the `payload` in line 66 to have orthorectify  on the "primary layer"?

Reply:
I have tried using `updateProcessingGetMapPayload` in a separate branch, but there are two problems with it:
- it is `protected` (and I'm reluctant to change it to `public`)
- it assumes that there is only a single dataset

In the end the branch touched too many places and I wasn't satisfied with how it looked, so I scraped it. I do agree with you though and would like to rearrange the code again now that there is a bit more time for it.